src/main/java/com/exasol/sql/expression/rendering/ValueExpressionRenderer.java

```diff
@@ -14,53 +15,113 @@ public ValueExpressionRenderer(final StringRendererConfig config) {
 
     @Override
     public void visit(final StringLiteral literal) {
+        appendCommaWhenNeeded(literal);
         append("'");
         append(literal.toString());
         append("'");
+        setLastVisited(literal);
     }
 
     @Override
     public void visit(final IntegerLiteral literal) {
+        appendCommaWhenNeeded(literal);
         append(literal.toString());
+        setLastVisited(literal);
     }
 
     @Override
     public void visit(final LongLiteral literal) {
+        appendCommaWhenNeeded(literal);
         append(literal.toString());
+        setLastVisited(literal);
     }
 
     @Override
     public void visit(final DoubleLiteral literal) {
+        appendCommaWhenNeeded(literal);
         append(literal.toString());
+        setLastVisited(literal);
     }
 
     @Override
     public void visit(final FloatLiteral literal) {
+        appendCommaWhenNeeded(literal);
         append(literal.toString());
+        setLastVisited(literal);
     }
 
     @Override
     public void visit(final BooleanLiteral literal) {
+        appendCommaWhenNeeded(literal);
         append(literal.toString());
+        setLastVisited(literal);
     }
 
     @Override
     public void visit(final ColumnReference columnReference) {
-        final String tableName = columnReference.getTableName();
-        if ((tableName != null) && !tableName.isEmpty()) {
-            append(tableName);
-            append(".");
-        }
-        append(columnReference.getColumnName());
+        appendCommaWhenNeeded(columnReference);
+        appendAutoQuoted(columnReference.toString());
+        setLastVisited(columnReference);
     }
 
     @Override
     public void visit(final UnnamedPlaceholder unnamedPlaceholder) {
         append("?");
+        setLastVisited(unnamedPlaceholder);
     }
 
     @Override
     public void visit(final DefaultValue defaultValue) {
         appendKeyword("DEFAULT");
+        setLastVisited(defaultValue);
+    }
+
+    @Override
+    public void visit(final Function function) {
+        appendCommaWhenNeeded(function);
+        appendKeyword(function.getFunctionName());
+        if (function.hasParenthesis()) {
+            startParenthesis();
+        }
+        setLastVisited(function);
+    }
+
+    @Override
+    public void leave(final Function function) {
+        if (function.hasParenthesis()) {
+            endParenthesis();
+        }
+        setLastVisited(function);
+    }
+
+    @Override
+    public void visit(final BinaryArithmeticExpression expression) {
+        startParenthesis();
+        appendOperand(expression.getLeft());
+        append(expression.getStringOperatorRepresentation());
+        appendOperand(expression.getRight());
+        endParenthesis();
+        setLastVisited(expression);
+    }
+
+    protected void appendOperand(final ValueExpression operand) {
+        final ValueExpressionRenderer leftExpressionRenderer = new ValueExpressionRenderer(this.config);
```

Comment:
Remove the "left" and I am happy. :-)
This part here works for both sides.

Reply:
fixed